include/ignition/math/Region3.hh

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#ifndef IGNITION_MATH_REGION3_HH_
+#define IGNITION_MATH_REGION3_HH_
+
+#include <cmath>
+#include <ostream>
+#include <utility>
+
+#include <ignition/math/Interval.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/config.hh>
+
+namespace ignition
+{
+  namespace math
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_MATH_VERSION_NAMESPACE {
+    //
+    /// \class Region3 Region3.hh ignition/math/Region3.hh
+    /// \brief The Region3 class represents the cartesian product
+    /// of intervals Ix ✕ Iy ✕ Iz, one per axis, yielding an
+    /// axis-aligned region of R^3 space. It can be thought of as
+    /// an intersection of halfspaces. Regions may be open or
+    /// closed in their boundaries, if any.
```

Comment:
I can imagine users being confused as to whether they should use this class or `AxisAlignedBox`. Do you think it's worth explaining why one would choose one over the other?

Reply:
Done in d7c74a5d0b181932eae68f6313184636ac0e8b52.